include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h

```diff
@@ -114,6 +114,80 @@ struct sycl_iterator
         return *this - it < 0;
     }
 
+    bool
```

Comment:
I think that relying on
`using iterator_category = ::std::random_access_iterator_tag;`
is enough. The iterator should not  be a "honest" random_access_iterator concept, IMHO. Also, oneAPI spec and doc doesn't require that.

Reply:
It is not enough, because C++20 allows defining the type of iterator using `iterator_concept` type alias. If we would check for `iterator_category` only, all "modern" iterators (in particular from C++ ranges) would be threated as input iterators.